utils/sql/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -925,7 +926,15 @@ public RuleType map(ResultSet rs, StatementContext ctx) throws SQLException {
                             }
                         })
                         .list();
+                if (rules.isEmpty()) {
+                    //TODO replace with a faster existence check
+                    //this checks if artifact actually exists and can throw ArtifactNotFoundException
+                    getArtifactVersions(artifactId);
+                }
+                return rules;
```

Comment:
Interesting solution!  Seems fine for now, but perhaps we can use the fast existence check I've created in the ksql impl here (in the future).